In [36]:
%history

from astropy.io import fits 
import matplotlib.pyplot as plt
import astropy 
import numpy as np 
from flow import flowLCT
%matplotlib inline
cube = fits.getdata('/home/oan1803/Documents/MOAT/Cubos/cubealign.fits')

#cube_filtered = ('/home/oan1803/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

#cube_filtered = ('/home/oan1803/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
cube.shape
vx = []
vy = []
vz = []

for i in range(15,838,15):
    velcube = flowLCT(cube[i-15:i], 2, 0.504, 720)
    vx.append(velcube.vx)
    vy.append(velcube.vy)
    vz.append(velcube.vz)
    #Para i=0 se tiene el intervalo de 0-15 y así sucesivamente. Hay un total de 55 intervalos de 0 a 838 imágenes.
print(vx[1])
print(len(vx))
velcube15 = flowLCT(cube[0:15], 2, 0.504, 720)
velcube30 = flowLCT(cube[15:30], 2, 0.504, 720)
velcube

In [37]:
def rescue_code(cube):
    import inspect
    get_ipython().set_next_input("".join(inspect.getsourcelines(cube)[0]))

In [42]:
rescue_code(clip)

NameError: name 'clip' is not defined

In [ ]:
def flowLCT(mc, fwhm_arcsec, scale, cadence,verbose=False, **kwargs):
    """
    Spanish: Programa para generar los mapas de flujos vectoriales vx, vy, vz  en km/s.
    English: Script to generate the vector flow maps vx, vy, vz in km/s.
    Parameters
    ----------
        mc :
            A `~numpy array` of shape ``(nt, ny,nx)`` where ``nt`` is the number of
            layers or images to calculate the flow maps.

        fwhm_arcsec :
            Window size for apodization (arcsec)
        scale :
            Size of pixel (Instument information)
        cadence:
            temporal sampling interval in the time series (seconds between image)

    kwargs flowmaker :
	----------------
	    method : {'square' | 'absolute' | 'cross'}
	    interpolation : {'fivepoint' | 'qfit2' | 'crossD'}
	    window : {'gaussian' | 'boxcar'}
		rebine: the rebinning factor if it is wished.
		lag: he lag between the images to be compared (number of images)

    Output
    -------
        The function returns the velocity maps for vx, vy, and vz, in km/s. For
        solar images cases, vz = h_m * div(vx,vy). For mor information check
        divergence function inside math_tools.py
    """
    reb = 1
    structure = OrderedDict()
    
    fwhm = fwhm_arcsec / scale
    
    kmperasec = 725  # Value of kilometers per arcsec'
    h_m = 150  # input('mass-flux scale-heigth (November 1989, ApJ,344,494):')
    
    delta_t = cadence  # time-lag in seconds
    factor = scale * kmperasec / delta_t
    v_limit = 2*reb +reb  # cota maxima velocidad en pixels.
    
    # ************************************************************
    #               Applying LCT
    # ************************************************************
    
 
    HorizontalVelocities = pyflowmaker(mc, fwhm, **kwargs)
    vx = HorizontalVelocities.vx
    vy = HorizontalVelocities.vy
    
    vx = vx.clip(-v_limit, v_limit)
    vy = vy.clip(-v_limit, v_limit)
    
    vx_kps = vx * factor  # vx in km/s
    vy_kps = vy * factor  # vy in km/s
    
    vx_kps = vx_kps - np.mean(vx_kps)
    vy_kps = vy_kps - np.mean(vy_kps)
    
    div = divergence(vx_kps, vy_kps)
    
    vz_kps = h_m * div
    
    structure['vx'] = vx_kps
    structure['vy'] = vy_kps
    structure['vz'] = vz_kps

    FlowStructure = namedtuple('FlowStructure', sorted(structure))
    
    return FlowStructure(**structure)


from astropy.io import fits 
import matplotlib.pyplot as plt
import astropy 
import numpy as np 
from flow import flowLCT
%matplotlib inline
cube = fits.getdata('/home/oan1803/Documents/MOAT/Cubos/cubealign.fits')

#cube_filtered = ('/home/oan1803/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

#cube_filtered = ('/home/oan1803/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
cube.shape
vx = []
vy = []
vz = []

for i in range(15,838,15):
    velcube = flowLCT(cube[i-15:i], 2, 0.504, 720)
    vx.append(velcube.vx)
    vy.append(velcube.vy)
    vz.append(velcube.vz)
    #Para i=0 se tiene el intervalo de 0-15 y así sucesivamente. Hay un total de 55 intervalos de 0 a 838 imágenes.
print(vx[1])
print(len(vx))
velcube15 = flowLCT(cube[0:15], 2, 0.504, 720)
velcube30 = flowLCT(cube[15:30], 2, 0.504, 720)
velcube45 = flowLCT(cube[30:45], 2, 0.504, 720)
velcube60 = flowLCT(cube[45:60], 2, 0.504, 720)
velcube75 = flowLCT(cube[60:75], 2, 0.504, 720)


#vx = velcube15.vx
#vy = velcube15.vy

print(np.min(vx))
print(np.mean(vx))
print(np.max(vx))
#print(velx)
vx = []
vy = []
vz = []

for i in range(15,838,15):
    velcube = flowLCT(cube[i-15:i], 2, 0.504, 720)
    vx.append(velcube.vx)
    vy.append(velcube.vy)
    vz.append(velcube.vz)
    #Para i=0 se tiene el intervalo de 0-15 y así sucesivamente. Hay un total de 55 intervalos de 0 a 838 imágenes.
print(vx[1])
print(len(vx))
magnitud = np.sqrt((u)**2 + (v)**2)
M=np.max(magnitud)

np.std(u)
np.mean(v)
np.std(v)

print(M)
print(np.std(u))

n, b, _ = plt.hist(np.ravel(magnitud), bins=80)
plt.axvline(1.2, color='r')
uu = np.clip(u,-1,1)
vv = np.clip(v,-1,1)
print(np.max(uu))
#Se crean las matrices para generar la red en la que se van a dibujar las flechas

x = np.arange(0,277)
y = np.arange(0,277)

ejex=x*0.504
ejey=y*0.504

X, Y = np.meshgrid(ejex, ejey)  #Matrices X y Y que van de 0 a 277 en sus filas y columnas

u = vx[0]  #Matrices que van a definir la dirección de la flecha
v = vy[0]

uu = np.clip(u,-1,1) #Corte de velocidades(Velocidades mayores o menores a +-1 las convierte en 1)
vv = np.clip(v,-1,1)
print(np.max(uu))

# Se hace la red de flechas de las velocidades 

#color = np.sqrt((u)**2 + (v)**2)

fig, ax = plt.subplots(figsize=(14,14))

Q= ax.quiver(X[::3,::3],Y[::3,::3],uu[::3,::3],vv[::3,::3], color='navy', alpha=10, scale=25, headwidth= 4, headlength=4, width=0.0012)
plt.imshow(cube[7],cmap='gray',origin='lower',extent=(X.min(),X.max(),Y.min(),Y.max()))
magnitud = np.sqrt((u)**2 + (v)**2)
M=np.max(magnitud)

np.std(u)
np.mean(v)
np.std(v)

print(M)
print(np.std(u))

n, b, _ = plt.hist(np.ravel(magnitud), bins=80)
plt.axvline(1.2, color='r')
uu = np.clip(u,-1,1)
vv = np.clip(v,-1,1)
print(np.max(uu))
#Se crean las matrices para generar la red en la que se van a dibujar las flechas

x = np.arange(0,277)
y = np.arange(0,277)

ejex=x*0.504
ejey=y*0.504

X, Y = np.meshgrid(ejex, ejey)  #Matrices X y Y que van de 0 a 277 en sus filas y columnas

u = vx[0]  #Matrices que van a definir la dirección de la flecha
v = vy[0]

uu = np.clip(u,-1,1) #Corte de velocidades(Velocidades mayores o menores a +-1 las convierte en 1)
vv = np.clip(v,-1,1)
print(np.max(uu))

# Se hace la red de flechas de las velocidades 

#color = np.sqrt((u)**2 + (v)**2)

fig, ax = plt.subplots(figsize=(14,14))

Q= ax.quiver(X[::3,::3],Y[::3,::3],uu[::3,::3],vv[::3,::3], color='navy', alpha=10, scale=25, headwidth= 4, headlength=4, width=0.0012)
plt.imshow(cube[7],cmap='gray',origin='lower',extent=(X.min(),X.max(),Y.min(),Y.max()))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

#cube_filtered = ('/home/oan1803/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube_filtered = ('/home/oan1803/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = ('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = ('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[19,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = ('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[1,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = ('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[2], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = ('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[2,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = ('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

#plt.imshow(cube[2,:,:], cmap='gray')
print(type(cube))
cube.shape
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube1 = ('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

#plt.imshow(cube[2,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

#plt.imshow(cube[2,:,:], cmap='gray')
print(type(cube))
#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[2,:,:], cmap='gray')
print(type(cube))
cube.shape
vx = []
vy = []
vz = []

for i in range(15,838,15):
    velcube = flowLCT(cube[i-15:i], 2, 0.504, 720)
    vx.append(velcube.vx)
    vy.append(velcube.vy)
    vz.append(velcube.vz)
    #Para i=0 se tiene el intervalo de 0-15 y así sucesivamente. Hay un total de 55 intervalos de 0 a 838 imágenes.
print(vx[1])
print(len(vx))
velcube15 = flowLCT(cube[0:15], 2, 0.504, 720)
velcube30 = flowLCT(cube[15:30], 2, 0.504, 720)
velcube45 = flowLCT(cube[30:45], 2, 0.504, 720)
velcube60 = flowLCT(cube[45:60], 2, 0.504, 720)
velcube75 = flowLCT(cube[60:75], 2, 0.504, 720)


#vx = velcube15.vx
#vy = velcube15.vy

print(np.min(vx))
print(np.mean(vx))
print(np.max(vx))
#print(velx)
Promedio_cube = np.mean(cube[:15],axis=0)
print(Promedio_cube.shape)

plt.imshow(Promedio_cube, cmap='gray', origin= 'lower')
#plt.imshow(cube[7], cmap='gray')
magnitud = np.sqrt((u)**2 + (v)**2)
M=np.max(magnitud)

np.std(u)
np.mean(v)
np.std(v)

print(M)
print(np.std(u))

n, b, _ = plt.hist(np.ravel(magnitud), bins=80)
plt.axvline(1.2, color='r')
uu = np.clip(u,-1,1)
vv = np.clip(v,-1,1)
print(np.max(uu))
#Se crean las matrices para generar la red en la que se van a dibujar las flechas

x = np.arange(0,277)
y = np.arange(0,277)

ejex=x*0.504
ejey=y*0.504

X, Y = np.meshgrid(ejex, ejey)  #Matrices X y Y que van de 0 a 277 en sus filas y columnas

u = vx[0]  #Matrices que van a definir la dirección de la flecha
v = vy[0]

uu = np.clip(u,-1,1) #Corte de velocidades(Velocidades mayores o menores a +-1 las convierte en 1)
vv = np.clip(v,-1,1)
print(np.max(uu))

# Se hace la red de flechas de las velocidades 

#color = np.sqrt((u)**2 + (v)**2)

fig, ax = plt.subplots(figsize=(14,14))

Q= ax.quiver(X[::3,::3],Y[::3,::3],uu[::3,::3],vv[::3,::3], color='navy', alpha=10, scale=25, headwidth= 4, headlength=4, width=0.0012)
plt.imshow(cube[7],cmap='gray',origin='lower',extent=(X.min(),X.max(),Y.min(),Y.max()))
cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubealign.fits')

#cube = fits.getdata('/Volumes/VALERIA/AUSUS/Documents/MOAT/Cubos/cubo_filtered.fits')

plt.imshow(cube[2,:,:], cmap='gray')
print(type(cube))
cube.shape
%history -g
%history